src/ngMock/angular-mocks.js

```diff
@@ -556,9 +556,13 @@ angular.mock.$IntervalProvider = function() {
      */
     $interval.flush = function(millis) {
       now += millis;
-      while (repeatFns.length && repeatFns[0].nextTime <= now) {
+      var firstTask = repeatFns[0];
+      while (repeatFns.length && (repeatFns[0].nextTime <= now || !repeatFns[0].nextTime)) {
         var task = repeatFns[0];
         task.fn();
+        if (!task.delay) {
+          task.delay = task.nextTime = 1;
```

Comment:
Hey thanks! I was afraid of bothering you 😄 
It should be ready now, please let me know if I'm missing something else.